dm/worker/task_checker.go

```diff
@@ -90,21 +91,47 @@ func (bs ResumeStrategy) String() string {
 	return fmt.Sprintf("unsupported resume strategy: %d", bs)
 }
 
+// duration is used to hold a time.Duration field
+type duration struct {
+	time.Duration `toml:"duration" json:"duration"`
+}
+
+// MarshalText hacks to satisfy the encoding.TextMarshaler interface
+func (d duration) MarshalText() ([]byte, error) {
+	return []byte(d.Duration.String()), nil
+}
+
+// UnmarshalText hacks to satisfy the encoding.TextUnmarshaler interface
+func (d duration) UnmarshalText(text []byte) error {
+	var err error
+	d.Duration, err = time.ParseDuration(string(text))
+	return err
+}
+
+// MarshalJSON hacks to satisfy the json.Marshaler interface
+func (d *duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Duration string `json:"duration"`
```

Comment:
we have JSON flag in L96, is this still needed?

Reply:
yes, the JSON tag here is must, which is really used during JSON marshal

Reply:
can we remove this `MarshalJSON` method and only rely on the JSON flag in L96?

Reply:
If we don't implement this method, the JSON output will be `int64` value of `Duratioin`  in nanosecond level, which is not human readable